main.py

```
@@ -0,0 +1,45 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import List
+
+import crud
+from schemas import (Author, AuthorCreate, BookCreate, Book)
+
+from db.database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/authors/", response_model=Author)
+def create_author(author: AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(db, author)
+
+@app.get("/authors/", response_model=List[Author])
+def read_authors(skip: int = 0, limit: int = 10,
+                 db: Session = Depends(get_db)):
+    return crud.get_authors(db, skip, limit)
+
+@app.get("/authors/{author_id}", response_model=Author)
+def read_author(author_id: int, db: Session = Depends(get_db)):
+    return crud.get_author(db, author_id)
```

Issue: The `read_author` endpoint directly returns the result of `crud.get_author`, which may be `None` if the author does not exist. This will cause FastAPI to return a 200 response with a null body, or potentially a validation error. You should check if the author exists and raise an `HTTPException(status_code=404, detail='Author not found')` if not. This is a critical error related to proper API error handling.